src/main/java/org/radarcns/management/domain/MetaToken.java

```diff
@@ -0,0 +1,135 @@
+package org.radarcns.management.domain;
+
+import java.time.ZonedDateTime;
+import java.util.Objects;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.EntityListeners;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.SequenceGenerator;
+import javax.persistence.Table;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Pattern;
+
+import org.apache.commons.lang3.RandomStringUtils;
+import org.hibernate.annotations.Cache;
+import org.hibernate.annotations.CacheConcurrencyStrategy;
+import org.hibernate.envers.Audited;
+import org.radarcns.auth.config.Constants;
+import org.radarcns.management.domain.support.AbstractEntityListener;
+
+@Entity
+@Audited
+@Table(name = "radar_meta_token")
+@Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
+@EntityListeners({AbstractEntityListener.class})
+public class MetaToken extends AbstractEntity {
+
+
+    //https://math.stackexchange.com/questions/889538/
+    // probability-of-collision-with-randomly-generated-id
+    // Current length of tokenName is 12. If we think there might be collision we can increase
+    // the length.
```

Comment:
There are almost 6 bits of entropy per character (A-Za-z0-9 space size is 62 is close to 2^6). So there are 72 bits of entropy for the entire string. That is almost 5000 * billion * billion possibilities. For the number of users we're looking at, it should be enough for security. However, we still have to do collision detection.